hub-saml/src/main/java/uk/gov/ida/saml/hub/validators/response/idp/components/EncryptedResponseFromIdpValidator.java

```diff
@@ -27,7 +27,7 @@
 import static uk.gov.ida.saml.core.errors.SamlTransformationErrorFactory.unexpectedNumberOfAssertions;
 import static uk.gov.ida.saml.security.validators.signature.SamlSignatureUtil.isSignaturePresent;
 
-public class EncryptedResponseFromIdpValidator<T extends Enum> {
+public class EncryptedResponseFromIdpValidator<T extends Enum<?>> {
```

Comment:
Is there any benefit of using `<T extends Enum<?>>` instead of `<T extends Enum<T>>`?

Reply:
Yes - this avoids unchecked cast exceptions by using Java generics "properly"